### PR TITLE
Add markdown parser and HTML sanitizer to markdown panel

### PR DESCRIPTION
**Pull request by cndonovan**
Not a lot of code, but a lot of thought 🤔 

I tried a number of the major markdown parsing libraries in Javascript. I started off trying the libraries that Luke had surfaced, then tried some others. 

Main requirements are: 
- make sure to sanitize the HTML (i.e. convert text to HTML, _then_ sanitize) 
- make sure we can support Github Flavored Markdown (GFM adds support for tables and code-blocks and is also what most developers will be used to)
- lastly, general checks like "is this library well-maintained?" and "is this library a reasonable size?"

The best options came out as:
* `react-markdown`, wrapper for chaining a bunch of plugins in the `remark` ecosystem
* `micromark`, the markdown parser in the `remark` ecosystem
* `marked`, popular markdown parser that's been around for awhile

Ultimately, went with `marked` because it seems as good as `micromark` in terms of parsing markdown, and the sanitization step is more explicit.

### CODE REVIEW

**Comment by Nexucis** [comment]
👍 looks pretty cool :).

> make sure to sanitize the HTML (i.e. convert text to HTML, then sanitize)

Is it what `DomPurified` is doing ?

**Comment by Nexucis** [comment]
> Is it what DomPurified is doing ?

Alright forget my silly question I just missed this line : 

```typescript
const sanitizedHTML = DOMPurify.sanitize(html);
```

**Comment by Nexucis** [comment]
would be cool to see the cuelang definition too ;) (just saying :p)

**Comment by cndonovan** [comment]
> > Is it what DomPurified is doing ?
> 
> Alright forget my silly question I just missed this line :
> 
> ```ts
> const sanitizedHTML = DOMPurify.sanitize(html);
> ```

ya :)

**Comment by cndonovan** [comment]
> would be cool to see the cuelang definition too ;) (just saying :p)

haha yes good reminder

**Review by saminzadeh — APPROVED** [review]
LGTM

**Review by LukeTillman — APPROVED** [review]
Looks great, one minor nit inline. Thanks for knocking this out!

**Comment by LukeTillman** [comment]
Also, interested to see where you land with styling the markup. LMK if you need to brainstorm.

File: ui/panels-plugin/src/plugins/markdown/MarkdownPanel.tsx [hunk 1]
```diff
@@ -11,18 +11,27 @@
...
+
 export function MarkdownPanel(props: MarkdownPanelProps) {
   const {
     definition: {
       options: { text },
     },
   } = props;
 
-  return <Box sx={{ height: '100%', overflowY: 'auto' }}>{text}</Box>;
+  const html = marked.parse(text, { gfm: true });
```
Comment (eunicorn) [inline ui/panels-plugin/src/plugins/markdown/MarkdownPanel.tsx:33]: what does gfm mean?

File: ui/panels-plugin/src/plugins/markdown/MarkdownPanel.tsx [hunk 2]
```diff
@@ -11,18 +11,27 @@
...
+  const sanitizedHTML = DOMPurify.sanitize(html);
+
+  return <Box sx={{ height: '100%', overflowY: 'auto' }} dangerouslySetInnerHTML={{ __html: sanitizedHTML }} />;
```
Comment (eunicorn) [inline ui/panels-plugin/src/plugins/markdown/MarkdownPanel.tsx:36]: cool to see how `dangerouslySetInnerHTML` is used safely 😄

**Review by eunicorn — APPROVED** [review]

File: ui/panels-plugin/src/plugins/markdown/MarkdownPanel.tsx (hunk 1 above)
Comment (eunicorn) [reply ui/panels-plugin/src/plugins/markdown/MarkdownPanel.tsx:33]: oh never mind, i should read the description first haha `Github Flavored Markdown`

File: ui/panels-plugin/src/plugins/markdown/MarkdownPanel.tsx (hunk 2 above)
Comment (LukeTillman) [reply ui/panels-plugin/src/plugins/markdown/MarkdownPanel.tsx:36]: I was kinda jealous she got to use it before I did. 😂

Comment (cndonovan) [reply ui/panels-plugin/src/plugins/markdown/MarkdownPanel.tsx:36]: living on the edge :)

File: ui/panels-plugin/src/plugins/markdown/MarkdownPanel.tsx [hunk 3]
```diff
@@ -11,18 +11,27 @@
...
 export function MarkdownPanel(props: MarkdownPanelProps) {
   const {
     definition: {
       options: { text },
     },
   } = props;
 
-  return <Box sx={{ height: '100%', overflowY: 'auto' }}>{text}</Box>;
+  const html = marked.parse(text, { gfm: true });
+  const sanitizedHTML = DOMPurify.sanitize(html);
```
[the remark this answers is not in the document]
Comment (cndonovan) [reply ui/panels-plugin/src/plugins/markdown/MarkdownPanel.tsx:34]: sounds like a good idea!

File: ui/panels-plugin/src/plugins/markdown/MarkdownPanel.tsx (hunk 1 above)
Comment (cndonovan) [reply ui/panels-plugin/src/plugins/markdown/MarkdownPanel.tsx:33]: it sounds slightly like `lfg` 😂

**Pull request merged**